lib/db/mysql/patches/patch-014-015.sql

```diff
@@ -0,0 +1,50 @@
+-- Add purge expired tokens stored procedure and index to expiresAt column
+ALTER TABLE tokens ADD INDEX idx_expiresAt (expiresAt), ALGORITHM = INPLACE, LOCK = NONE;
+
+# Create procedure that will delete expired access tokens.
+#
+# -- (IN) deleteTokenCount : Deleted this amount of tokens. 1000 tokens are deleted each round.
+# -- (IN) delaySeconds : After every 1000 deletes, delay the next delete by this
+# -- (IN) ignoreClientId: Skip access tokens from this client
+# -- (OUT) totalDeleted: Number deleted tokens
+# -- (OUT) rounds: Number to iterations taken to delete tokens
+#
+#   Ex.
+#       CALL PURGE_EXPIRED_TOKENS(1000,0,'37e7061dd45df02f',@totalDeleted,@rounds);
+#       SELECT @totalDeleted, @rounds;
+#
+CREATE PROCEDURE PURGE_EXPIRED_TOKENS(IN deleteTokenCount INT, IN delaySeconds INT, IN ignoreClientId CHAR(16), OUT totalDeleted INT, OUT rounds INT)
```

Comment:
So...the more I learn about stored procedures, the less I feel like adding more of them to our codebase.  My latest gripe:  they apparently operate with the strict-mode setting in force at the time they were created, not the strict-mode in force at the time they're invoked.

How would you feel above moving this looping logic directly into javascript in the `db.purgeExpiredTokens` method?